osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/infra/networking/services/commands/Iec61850SetScheduleCommand.java

```diff
@@ -120,6 +125,28 @@ public Void apply(final DeviceMessageLog deviceMessageLog) throws ProtocolAdapte
                     return null;
                 }
 
+                private void writeAstronomicalOffsetsForSchedule(final DeviceMessageLog deviceMessageLog)
+                        throws NodeException {
+                    if (relayType.equals(RelayTypeDto.LIGHT) && astronomicalSunriseOffset != null
+                            && astronomicalSunsetOffset != null) {
+                        final NodeContainer softwareConfiguration = deviceConnection.getFcModelNode(
+                                LogicalDevice.LIGHTING, LogicalNode.STREET_LIGHT_CONFIGURATION,
+                                DataAttribute.SOFTWARE_CONFIGURATION, Fc.CF);
+
+                        softwareConfiguration.writeShort(SubDataAttribute.ASTRONOMIC_SUNRISE_OFFSET,
+                                astronomicalSunriseOffset);
+                        deviceMessageLog.addVariable(LogicalNode.STREET_LIGHT_CONFIGURATION,
+                                DataAttribute.SOFTWARE_CONFIGURATION, Fc.CF, SubDataAttribute.ASTRONOMIC_SUNRISE_OFFSET,
+                                Short.toString(astronomicalSunriseOffset));
+
+                        softwareConfiguration.writeShort(SubDataAttribute.ASTRONOMIC_SUNSET_OFFSET,
+                                astronomicalSunsetOffset);
+                        deviceMessageLog.addVariable(LogicalNode.STREET_LIGHT_CONFIGURATION,
+                                DataAttribute.SOFTWARE_CONFIGURATION, Fc.CF, SubDataAttribute.ASTRONOMIC_SUNSET_OFFSET,
+                                Short.toString(astronomicalSunsetOffset));
+                    }
+                }
+
                 private void writeScheduleEntryForRelay(final DeviceMessageLog deviceMessageLog,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "writeScheduleEntryForRelay" is 11 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)